samples_project/Assets/SampleViewer/Samples/VRSample/Scripts/ContinuousMovement.cs

```diff
@@ -28,25 +62,74 @@ void Start()
     private void FixedUpdate()
     {
         FollowHeadset();
-        Quaternion headYaw = Quaternion.Euler(0, rig.Camera.transform.eulerAngles.y, 0);
-        Vector3 direction = headYaw * new Vector3(leftInputAxis.x, 0, leftInputAxis.y);
-        Vector3 up = new Vector3(0, rightInputAxis.y, 0);
-        controller.Move(direction * speed * Time.fixedDeltaTime);
-        controller.Move(up * upSpeed * Time.fixedDeltaTime);
+        var headYaw = Quaternion.Euler(0, rig.Camera.transform.eulerAngles.y, 0);
+        var direction = headYaw * new Vector3(leftInputAxis.x, 0, leftInputAxis.y);
+        var up = new Vector3(0, rightInputAxis.y, 0);
+        controller.Move(speed * Time.fixedDeltaTime * direction);
+        controller.Move(upSpeed * Time.fixedDeltaTime * up);
+    }
 
+    private void LateUpdate()
+    {
+        UICanvas.transform.position = playerCamera.transform.position + playerCamera.transform.forward * 15;
+        UICanvas.transform.rotation = new Quaternion(0.0f, playerCamera.transform.rotation.y, 0.0f, playerCamera.transform.rotation.w);
     }
     // Update is called once per frame
     void Update()
     {
-        InputDevice leftDevice = InputDevices.GetDeviceAtXRNode(leftInputSource);
+        var leftDevice = InputDevices.GetDeviceAtXRNode(leftInputSource);
         leftDevice.TryGetFeatureValue(CommonUsages.primary2DAxis, out leftInputAxis);
-        InputDevice rightDevice = InputDevices.GetDeviceAtXRNode(rightInputSource);
+        var rightDevice = InputDevices.GetDeviceAtXRNode(rightInputSource);
         rightDevice.TryGetFeatureValue(CommonUsages.primary2DAxis, out rightInputAxis);
+        if (LeftMenuAction.triggered)
+        {
+            ToggleCanvas();
+        }
+        if (RightMenuAction.triggered)
+        {
+            ToggleCanvas();
+        }
     }
     void FollowHeadset()
     {
         controller.height = rig.CameraInOriginSpaceHeight + heightOffset;
-        Vector3 capsuleCenter = transform.InverseTransformPoint(rig.Camera.transform.position);
+        var capsuleCenter = transform.InverseTransformPoint(rig.Camera.transform.position);
         controller.center = new Vector3(capsuleCenter.x, controller.height / 2 + controller.skinWidth, capsuleCenter.z);
     }
+
+    public void ToggleSmoothTurn()
```

Comment:
Just checking, is this func and `ToggleCanvas()` called by other scripts? If not these should be private

Reply:
They are called in through the event system on buttons/check box toggles. I'll test them as private but I don't believe they will be accessible in the event system if so.

Reply:
They need to be public to be called through the event system.